File: new_practice_cpp/1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        map<int,int> dict;
        map<int,int>::iterator iter;
        vector<int> ans;

        for (int i=0; i<nums.size();i++){
            
            iter = dict.find(target-nums[i]);
            // cout<<iter;
            // printf("(%d,%d)",iter->first,iter->second);
            if (iter != dict.end()){ // num found
                ans.push_back(iter->second); //nums find in dict (smaller)
                ans.push_back(i); // current index (bigger)                
                return ans;     
            }
            else{ //iter == dict.end(), not found
                dict[nums[i]]=i;    
            }
        }
        return ans;
    }
};
// @lc code=end

File: new_practice_python/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum

#  nums : List[int]
#  target: int

# List[int]

# @lc code=start


class Solution:
    def twoSum(self, nums, target): # 2 7 11 15
        # for idx, ele in enumerate(nums):
        #     # print(idx, ele) # 0 2
        #     if target - ele in nums[idx+1:]:
        #         if target - ele == ele:
        #             return [nums.index(ele), (nums[idx+1:].index(target - ele))+idx+1]
        #         else:
        #             return [nums.index(ele), nums.index(target - ele)]


        #     else:
        #         pass

        my_dict = {}
        for idx, ele in enumerate(nums):
            rest = target - ele
            if rest in my_dict:
                return [my_dict[rest], idx]
            else:
                my_dict[ele] = idx # record




       
# @lc code=end

